exetera/core/fields.py

```diff
@@ -495,27 +537,55 @@ def __getitem__(self, item):
         :return: Item value from dataset
         """
         if isinstance(item, slice):
-            start = item.start if item.start is not None else 0
-            stop = item.stop if item.stop is not None else len(self._indices) - 1
-            step = item.step
-            # TODO: validate slice
-            index = self._indices[start:stop + 1]
-            bytestr = self._values[index[0]:index[-1]]
-            results = [None] * (len(index) - 1)
-            startindex = self._indices[start]
-            for ir in range(len(results)):
-                results[ir] = \
-                    bytestr[index[ir] - np.int64(startindex):
-                            index[ir + 1] - np.int64(startindex)].tobytes().decode()
-            return results
+            if self._field_instance.filter is None:
+                start = item.start if item.start is not None else 0
+                stop = item.stop if item.stop is not None else len(self._indices) - 1
+                step = item.step
+                # TODO: validate slice
+                index = self._indices[start:stop + 1]
+                bytestr = self._values[index[0]:index[-1]]
+                results = [None] * (len(index) - 1)
+                startindex = self._indices[start]
+                for ir in range(len(results)):
+                    results[ir] = \
+                        bytestr[index[ir] - np.int64(startindex):
+                                index[ir + 1] - np.int64(startindex)].tobytes().decode()
+                return results
+            else:
+                mask = self._field_instance.filter[item]
+                if utils.is_sorted(mask):
+                    index_s = self._indices[mask]
```

Comment:
`# the filtered indices represent a filter operation`
We need to evaluate whether this saves anything. H5py is horrifically slow and the risk is that all of this work is lost when loading slices through its api. I would suggest just loading all of the data and applying the filtered index rather than trying to gain time here, unless we do a series of detailed benchmarks that can give us a heuristic to decide whether to do this or not